### PR TITLE
Turn on -Wextra/-Werror. Fix resulting errors.

### PR DESCRIPTION
**Pull request by Jeff-Ciesielski**
There were a couple of existing warnings in the code with the current build settings.  Ratcheting up the warning level unmasked a few more minor issues that shouldn't have any effect on the runtime behavior.

Figured now is a good time to do this as we're going to be hacking on grbl with increasing frequency.

### CODE REVIEW

File: system.h [hunk 1]
```diff
@@ -161,12 +161,17 @@ enum {
 };
 
 #define ACTIVE_TIMER time_STEP_ISR
-#define TIME_OFF(tid)  (((tid)==ACTIVE_TIMER)?(TIMING_PORT|=TIMING_MASK):0)
-#define TIME_ON(tid)  (((tid)==ACTIVE_TIMER)?(TIMING_PORT&=~TIMING_MASK):0)
-#define TIME_TOGGLE(tid)  (((tid)==ACTIVE_TIMER)?(TIMING_PIN|=TIMING_MASK):0)
+#define TIME_OFF(tid)  (((tid)==ACTIVE_TIMER)?(TIMING_PORT|=TIMING_MASK):(void)tid)
```
Comment (mdicicco) [inline system.h:164]: Where is 'tid' defined?  I did a quick scan above and didn't see it.

Comment (Jeff-Ciesielski) [reply system.h:164]: It's the symbol that's passed into the macro

**Comment by penguinland** [comment]
LGTM! Thanks for turning those on.

File: system.h (hunk 1 above)
Comment (mdicicco) [reply system.h:164]: oh, I see.

**Pull request merged**